### PR TITLE
Fixes #32383: Set Pulp to expect Foreman host as the client authentic…

### PR DESCRIPTION
**Pull request by ehelms**
…ating

This sets Apache to expect the client certificate to contain as the
common name the hostname of Foreman. This corresponds to using the
Foreman client certificates to talk to Pulp's API which is expected
to contain the hostname of Foreman in the certificate.

### CODE REVIEW

**Comment by ehelms** [comment]
Will require https://github.com/theforeman/puppet-pulpcore/pull/186 in a release

**Comment by ekohl** [comment]
I've converted it to a draft to indicate it's blocked on the other PR.

**Comment by ehelms** [comment]
Moved out of draft mode, with the way this functionality works in puppet-pulpcore, the addition of this parameter here is an enhancement and does not break current implementations that us the certificates with CN set to admin. The only thing this needs is a release of or version bump of puppet-pulpcore.

File: manifests/init.pp
Comment (ekohl) [inline manifests/init.pp:258]:
```suggestion
    api_client_auth_cn             => {$foreman_host => 'admin'},
```

Comment (ekohl) [reply manifests/init.pp:258]: Reading more, there's no guarantee that the CN on the foreman host is the same as the client cert. I'm trying to decouple that. For example, it may be `foreman.example.com` but the host uses `foreman01.example.com` as the hostname. I'm not entirely sure what the right approach is today, but perhaps the Foreman Proxy's trusted host settings is a better indicator. That typically includes it.

Comment (ehelms) [reply manifests/init.pp:258]: The trusted hosts can be a list right? So are suggesting crafting a mapping of each trusted host entry to admin?

Comment (ekohl) [reply manifests/init.pp:258]: I'm not entirely sure. That's what I also thought about. I think it makes sense to treat everything that can connect to a Smart Proxy with full permissions also as something that can connect to the Pulp instance with full permissions.

**Review by ekohl — APPROVED** [review]

File: manifests/init.pp
Comment (ekohl) [inline manifests/init.pp:229]: I think you're not required to use a data type so this also works.
```suggestion
  $api_client_auth_cn_map = Hash($foreman_proxy::trusted_hosts.map |$host| {
```

However, it's fine to keep it in.

**Review by ekohl — APPROVED** [review]

**Comment by ekohl** [comment]
This is safe to merge now, right? All dependencies are in and this should be in before Katello's change is merged. It's also in addition to the current certs so doesn't break. If I'm right about that, please merge when you feel OK about it.

**Pull request merged**
